core/src/main/scala/org/bitcoins/core/wallet/utxo/InternalAddressTag.scala

```diff
@@ -0,0 +1,77 @@
+package org.bitcoins.core.wallet.utxo
+
+/**
+  * An AddressTagType that is native to Bitcoin-S.
+  * InternalAddressTagTypes are still usable when using Bitcoin-S
+  * as a dependency
+  */
+sealed trait InternalAddressTagType extends AddressTagType
+
+/**
+  * An AddressTag that is native to Bitcoin-S.
+  * InternalAddressTags are still usable when using Bitcoin-S
+  * as a dependency
+  */
+sealed trait InternalAddressTag extends AddressTag
+
+/** An unknown address tag type, should mostly be used for parsing external address tags */
+case class UnknownAddressTagType(name: String) extends InternalAddressTagType
+
+/** An address tag without an unknown type, should mostly be used for parsing external address tags */
+case class UnknownAddressTag(
+    name: String,
+    tagType: AddressTagType = UnknownAddressTagType(""))
+    extends InternalAddressTag
+
+object InternalAddressTagType {
+  val all: Seq[InternalAddressTagType] = Vector(StorageLocationTagType)
+
+  def fromString(string: String): Option[InternalAddressTagType] =
+    all.find(_.name.toLowerCase == string.toLowerCase)
+}
+
+object InternalAddressTag {
+
+  def fromUnknown(unknown: UnknownAddressTag): Option[InternalAddressTag] = {
+    InternalAddressTagType.fromString(unknown.tagType.name) match {
+      case Some(tagType) =>
+        tagType match {
+          case StorageLocationTagType =>
+            StorageLocationTag.fromString(unknown.name)
+        }
+      case None => None
+    }
+  }
+}
```

Comment:
When would this ever be used?

Reply:
You should have a `fromString` instead, and not use `Unknown` as an intermediate state

Reply:
I use `fromUnknown` because we need both the tag and the tag type to properly get an address tag, so it'd need to be `from(str1: String, str2: String)`.I think using `UnknownAddressTag` is just a better way of doing this